lib/schema.js

```diff
@@ -4,15 +4,32 @@ const Joi = require('joi');
 
 const internals = {};
 
+// internals.model = Joi.func().keys({
+//     tableName: Joi.string().required()
+// }).unknown();
 internals.model = Joi.func();
 
 exports.plugin = Joi.object({
+
+    knexFile: Joi.object(),
+    migration: Joi.object({
```

Comment:
I think it at least needs to be passed per plugin, since each plugin might have different models and in turn different migrations.  Yeah/no?